app/scripts/views/complete_sign_up.js

```diff
@@ -139,10 +141,21 @@ define(function (require, exports, module) {
             } else if (
                 AuthErrors.is(err, 'INVALID_VERIFICATION_CODE') ||
                 AuthErrors.is(err, 'INVALID_PARAMETER')) {
-              // These server says the verification code or any parameter is
-              // invalid. The entire link is damaged.
-              verificationInfo.markDamaged();
-              err = AuthErrors.toError('DAMAGED_VERIFICATION_LINK');
+
+              // When coming from sign-in confirmation verification, show a
+              // verification link expired error instead of damaged verification link.
+              // This error is generated because the link has already been used.
+              if (self.viewName === 'complete-signin') {
```

Comment:
This is where overloading the `complete-signup` view and `user.completeAccountSignUp` for both signup and signin shows its weakness. We have to special case on signup/signin in the view. My mistake. A more elegant solution, though outside of the scope of this PR, would be to create distinct views, one `complete-signup`, one `complete-signin`, as well as add corresponding user and account methods, and ditch the special casing here. `user.completeAccountSignIn` or `account.completeSignIn` could then check for `INVALID_VERIFICATION_CODE` or `INVALID_PARAMETER` and convert to `REUSED_SIGNIN_VERIFICATION_CODE` instead of mixing that into view code.